mutt/file.c

```diff
@@ -688,6 +688,50 @@ char *mutt_file_read_line(char *s, size_t *size, FILE *fp, int *line, int flags)
   }
 }
 
+/**
+ * mutt_file_iter_line - iterate over the lines from an open file pointer
+ * @param iter           state of iteration including ptr to line
+ * @param fp             file pointer to read from
+ * @param flags          same as mutt_file_read_line
+ * @retval               true iff data read, false on eof
+ *
+ * This is a slightly cleaner interface for mutt_file_read_line() which avoids
+ * the eternal C loop initialization ugliness.  Use like this:
+ *
+ * struct mutt_file_iter i = { 0 };
+ * while (mutt_file_iter_line(&i, fp, flags) {
+ *   do_stuff(i.s, i.line);
+ * }
+ */
+bool mutt_file_iter_line(struct mutt_file_iter *i, FILE *fp, int flags)
+{
+  char *p = mutt_file_read_line(i->s, &i->size, fp, &i->line, flags);
+  if (p == NULL)
+    return false;
+  i->s = p;
+  return true;
+}
+
+/**
+ * mutt_file_map_lines - a high level fun to process fp's lines as in Python etc
+ * @param f              function to call for each line
+ * @param d              arbitrary data passed to ``f''
+ * @param fp             file pointer to read from
+ * @param flags          same as mutt_file_read_line
+ * @retval               true iff all data mapped, false if ``f'' returns false
+ */
+bool mutt_file_map_lines(mutt_file_map_function f, void *d, FILE *fp, int flags)
+{
+  struct mutt_file_iter i = { 0 };
+  while (mutt_file_iter_line(&i, fp, flags)) {
+    if (!(*f)(i.s, i.line, d)) {
```

Comment:
This line is arcane to read. 

The variable names are too abstract and do not offer insight into what their purpose is. I can read the above documentation to understand what they do, but I'm forgetful and will need to reference the docs a few times :(

I'd recommend:
 - rename `f` -> `func`
 - rename `i` -> `iter`
 - rename `d` -> `data`

Reply:
Yes, good point.  Will do.

Reply:
See my latest push